tests/thread_tests/remote.go

```diff
@@ -0,0 +1,223 @@
+		remoteInfraInterface = v
+	}
+
+	if v := os.Getenv(remoteSnapPathEnv); v != "" {
+		remoteSnapPath = v
+	}
+
+	if v := os.Getenv(remoteGPIOChipEnv); v != "" {
+		remoteGPIOChip = v
+	}
+
+	if v := os.Getenv(remoteGPIOLineEnv); v != "" {
+		remoteGPIOLine = v
+	}
+}
+
+func connectSSH(t *testing.T) {
+	if SSHClient != nil {
+		return
+	}
+
+	config := &ssh.ClientConfig{
+		User: remoteUser,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(remotePassword),
+		},
+		Timeout:         10 * time.Second,
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+
+	var err error
+	SSHClient, err = ssh.Dial("tcp", remoteHost+":22", config)
+	if err != nil {
+		t.Fatalf("Failed to dial: %s", err)
+	}
+
+	t.Cleanup(func() {
+		SSHClient.Close()
+	})
+
+	t.Logf("SSH: connected to %s", remoteHost)
+}
+
+func remote_deployOTBRAgent(t *testing.T) {
+
+	t.Cleanup(func() {
+		remote_exec(t, "sudo snap remove --purge openthread-border-router")
+	})
+
+	start := time.Now().UTC()
+
+	commands := []string{
+		"sudo snap remove --purge openthread-border-router",
+		"sudo snap install openthread-border-router --edge",
+		"sudo snap set openthread-border-router infra-if='" + remoteInfraInterface + "'",
+		"sudo snap set openthread-border-router webgui-port=31190",
+		// "sudo snap connect openthread-border-router:avahi-control",
+		"sudo snap connect openthread-border-router:firewall-control",
+		"sudo snap connect openthread-border-router:raw-usb",
+		"sudo snap connect openthread-border-router:network-control",
+		// "sudo snap connect openthread-border-router:bluetooth-control",
+		// "sudo snap connect openthread-border-router:bluez",
+		"sudo snap start openthread-border-router",
+	}
+	for _, cmd := range commands {
+		remote_exec(t, cmd)
+	}
+
+	remote_waitForLogMessage(t, otbrSnap, "Start Thread Border Agent: OK", start)
+	t.Log("OTBR on remote device is ready")
+}
+
+func remote_deployGPIOCommander(t *testing.T) {
+
+	t.Cleanup(func() {
+		remote_exec(t, "sudo snap remove --purge "+matterGPIOSnap)
+	})
+
+	installCommand := fmt.Sprintf("sudo snap install %s --edge", matterGPIOSnap)
+	extraInterface := ""
+	if remoteSnapPath != "" {
+		installCommand = fmt.Sprintf("sudo snap install --dangerous %s", remoteSnapPath)
+		extraInterface = fmt.Sprintf("sudo snap connect %s:custom-gpio %s:custom-gpio-dev", matterGPIOSnap, matterGPIOSnap)
+	}
+
+	start := time.Now().UTC()
+
+	commands := []string{
+		fmt.Sprintf("sudo snap remove --purge %s ", matterGPIOSnap),
+		installCommand,
+		extraInterface,
+		fmt.Sprintf("sudo snap set %s args=\"--thread\"", matterGPIOSnap),
+		fmt.Sprintf("sudo snap set %s gpiochip=\"%s\"", matterGPIOSnap, remoteGPIOChip),
+		fmt.Sprintf("sudo snap set %s gpio=\"%s\"", matterGPIOSnap, remoteGPIOLine),
+		fmt.Sprintf("sudo snap connect %s:avahi-control", matterGPIOSnap),
+		fmt.Sprintf("sudo snap connect %s:otbr-dbus-wpan0 %s:dbus-wpan0", matterGPIOSnap, otbrSnap),
+		fmt.Sprintf("sudo snap start %s", matterGPIOSnap),
```

Comment:
Inconsistent coding style between 114-124 and 153-158 for the use of constant for snap's name. Think of readability and maintainability and choose one.